core/app/models/spree/stock_location.rb

```diff
@@ -12,6 +12,9 @@ class InvalidMovementError < StandardError; end
     belongs_to :state, class_name: 'Spree::State'
     belongs_to :country, class_name: 'Spree::Country'
 
+    has_many :shipping_method_stock_locations, class_name: Spree::ShippingMethodStockLocation
```

Comment:
As Ben says, this should be `dependent: :destroy`. Also, `class_name` should always be a string.